File: products/sdlc-cc/platform/packages/insights-core/go/scorer/scorer.go
package scorer

import (
	"github.com/sdlc-ai/platform/packages/insights-core/types"
)

// Scorer is a pure function; no I/O. Full impl lands in T-101.
type Scorer struct {
	Weights types.Weights
}

func New(w types.Weights) *Scorer { return &Scorer{Weights: w} }

// Score is a placeholder returning the weighted sum of the supplied breakdown
// clamped to [0,1]. T-101 replaces with the full formula + property tests.
func (s *Scorer) Score(b types.ScoreBreakdown) float64 {
	v := s.Weights.SOC2*b.SOC2 +
		s.Weights.HIPAA*b.HIPAA +
		s.Weights.GDPR*b.GDPR +
		s.Weights.Cost*b.Cost +
		s.Weights.Blast*b.Blast
	if v < 0 {
		return 0
	}
	if v > 1 {
		return 1
	}
	return v
}

File: products/sdlc-cc/platform/packages/insights-core/ts/src/scorer.ts
import type { ScoreBreakdown, Weights } from "./types.js";

export class Scorer {
  constructor(public readonly weights: Weights) {}

  score(b: ScoreBreakdown): number {
    const v =
      this.weights.soc2 * b.soc2 +
      this.weights.hipaa * b.hipaa +
      this.weights.gdpr * b.gdpr +
      this.weights.cost * b.cost +
      this.weights.blast * b.blast;
    if (v < 0) return 0;
    if (v > 1) return 1;
    return v;
  }
}
